File: backend/services/pdf_parser.py
import fitz

def extract_text_from_pdf(file_bytes: bytes) -> str:
    try:
        with fitz.open(stream=file_bytes, filetype="pdf") as doc:
            text = "".join(page.get_text() for page in doc)
        return text.strip()
    except Exception as e:
        print(f"PDF parse error: {e}")
        return ""

File: src/lib/pdf-parser.ts
import { extractText, getDocumentProxy } from 'unpdf';

export async function extractTextFromPdf(
    fileBytes: ArrayBuffer,
): Promise<string> {
    try {
        const pdf = await getDocumentProxy(new Uint8Array(fileBytes));
        const { text } = await extractText(pdf, { mergePages: true });
        return text.trim();
    } catch (error) {
        console.error('PDF parse error:', error);
        return '';
    }
}
